### ISSUE TITLE
Add hooks

### ISSUE THREAD

**Issue by imajus** [Lepozepo/S3#42]
I wonder if that's possible to hook intro package's server S3.upload function in order to alter the file before actual uploading. In my case I want to resize huge images before sending them to S3. Sure I can deny uploading files based on their size in bytes or mime type using FileList object data but I'd like to make it more user friendly. Also I believe it's not safe to wholly rely on client side code here.
I'm thinking of something like a new S3.config option like `onUpload: function(file) { }` which will either manage the `file` argument passed and returns true/false to allow/deny uploading OR will return changed `file` object / null.
Does it make any sense?

**Comment by timfam** [comment]
@Lepozepo we are using this package for https://github.com/orionjs hooks will be greatly appreciated. There just has to be a formalized way to do image resize and other manipulation. What say you? Thanks in advance

**Comment by Lepozepo** [comment]
I agree, once I get my Retail Management app into production I'll take a couple of weeks off to upgrade this package a little more ^_^

**Comment by timfam** [comment]
You built the retail app in meteor?

**Comment by Lepozepo** [comment]
Yep ^_^ it's specifically built for managing RMU's and small shops that are powered by Square (since I own one, lol)

**Comment by timfam** [comment]
Cool! I like to see it when it's ready you never know ;)

**Comment by Lepozepo** [comment]
Sweet! I'll send you a beta invite once I have it at that point ^_^

**Comment by timfam** [comment]
Looking forward to that!

**Comment by gibex** [comment]
I'm also looking for a similar solution (manipulate image before upload). 
Thanks.
